event/src/unvalidated/payload/data.rs

```diff
@@ -0,0 +1,87 @@
+use crate::unvalidated::{Value, ValueMap};
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+
+/// Payload of a data event
+#[derive(Serialize, Deserialize)]
+pub struct Payload<D> {
+    id: Cid,
+    prev: Cid,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    header: Option<Header>,
+    data: D,
+}
+
+impl<D> Payload<D> {
+    /// Construct a new payload for a data event
+    pub fn new(id: Cid, prev: Cid, header: Option<Header>, data: D) -> Self {
+        Self {
+            id,
+            prev,
+            header,
+            data,
+        }
+    }
+
+    /// Get the id
+    pub fn id(&self) -> &Cid {
+        &self.id
+    }
+
+    /// Get the prev
+    pub fn prev(&self) -> &Cid {
+        &self.prev
+    }
+
+    /// Get the header
+    pub fn header(&self) -> Option<&Header> {
+        self.header.as_ref()
+    }
+
+    /// Get a header value
+    pub fn header_value(&self, key: &str) -> Option<&Value> {
+        self.header.as_ref().and_then(|h| h.additional.get(key))
+    }
+
+    /// Get the data
+    pub fn data(&self) -> &D {
+        &self.data
+    }
+}
+
+/// Headers for a data event
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Header {
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub(crate) controllers: Vec<String>,
+    #[serde(flatten, skip_serializing_if = "ValueMap::is_empty")]
```

Comment:
I believe this will allow arbitrary fields to show up in the Header, which we don't want.  We should have an explicit list of fields that are allowed by the protocol to show up in the header.

Currently I believe that list is:
* controllers
* sep
* model
* unique
* shouldIndex